src/axom/primal/geometry/Polyhedron.hpp

```diff
@@ -601,11 +601,27 @@ class Polyhedron
   /*!
    * \brief Simple check for validity of a polyhedron
    *
-   * Initial check is that the polyhedron has four or more vertices
+   * Initial check is that the polyhedron has four or more unique vertices
    * \return True, if the polyhedron is valid, False otherwise
    */
   AXOM_HOST_DEVICE
-  bool isValid() const { return m_num_vertices >= 4; }
+  bool isValid() const
+  {
+    for(int i = 0; i < m_num_vertices; i++)
+    {
+      for(int j = i + 1; j < m_num_vertices; j++)
+      {
+        // operator= for Point does not want to play nice...
+        if(m_vertices[i][0] == m_vertices[j][0] &&
+           m_vertices[i][1] == m_vertices[j][1] &&
+           m_vertices[i][2] == m_vertices[j][2])
+        {
+          return false;
+        }
+      }
+    }
```

Comment:
>  do we always want to call a Polyhedron invalid when it has duplicated vertices?

Admittedly, I'm not sure.
I did find this case needs to be handled for the Polyhedron::volume() function. It resulted in an infinite loop condition.
I've added a new test in `primal_clip.cpp` for this case in particular.

>Also, should we be checking for exact duplicates? or should we be supplying a tolerance and checking that the distance b/w vertices is not nearly zero subject to that tolerance?

👍 . Agreed, a tolerance value would be good.

>Suggestion: I think it would be clearer to move the highlighted code to a new function hasDuplicatedVertices(double EPS), and consider calling it explicitly, when necessary.

I can separate this snippet into the function, and have it just in the `Polyhedron::volume()` function for now where I was originally seeing the problem.